synapse/media/media_repository.py

```diff
@@ -276,6 +277,14 @@ async def get_remote_media(
         ):
             raise FederationDeniedError(server_name)
```

Comment:
This makes me wonder if we would want to deny federation completely to these domains? Why just media?

Reply:
We want the ability to keep rooms functional without serving media, essentially. The federation deny list would prevent Synapse from otherwise keeping the room functional (at least in a performant way).

Reply:
In a business context this is a valid use case.  This protects against unwanted and unknown file transfers.